params/hooks_libevm.go

```diff
@@ -106,16 +107,19 @@ func makePrecompile(contract contract.StatefulPrecompiledContract) libevm.Precom
 		if err != nil {
 			panic(err) // Should never happen
 		}
-		var predicateResultsBytes []byte
-		if len(header.Extra) >= DynamicFeeExtraDataSize {
-			predicateResultsBytes = header.Extra[DynamicFeeExtraDataSize:]
+		var predicateResults *predicate.Results
+		if predicateResultsBytes := GetPredicateResultBytes(header.Extra); len(predicateResultsBytes) > 0 {
+			predicateResults, err = predicate.ParseResults(predicateResultsBytes)
+			if err != nil {
+				panic(err) // Should never happen, as results are already validated in block validation
+			}
 		}
 		accessableState := accessableState{
 			env: env,
 			blockContext: &BlockContext{
```

Comment:
why we cannot use vm.BlockContext directly from env? It should already have a parsed results (and other similar functions to implement required interface)

Reply:
perhaps you are looking at the subnet-evm or coreth source code?
as the vm.BlockContext here would be from the upstream libevm code [here](https://github.com/ava-labs/libevm/blob/main/core/vm/evm.go#L64), which we modified to include the full header so we can access the extra bits (aka predicate results) here and pass it to the precompiles.